main.go

```diff
@@ -64,12 +89,45 @@ func resyncAll(config *rest.Config) error {
 func main() {
 	flag.Parse()
 
+	// Create OS signal channels.
+	sigs := make(chan os.Signal, 1)
+	stop := make(chan struct{})
+	signal.Notify(sigs, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-sigs
+		glog.V(2).Info("Stopping informers")
+		close(stop)
```

Comment:
Unfortunately, SharedInformerFactory doesn't give us any way to wait for the informers to cleanly stop. Since we're about to terminate the process anyway, I don't think we gain anything by closing this channel. I suspect stopping was only implemented to avoid unbounded resource growth during tests that start and stop many informers within one process.

With that said, it doesn't hurt to close it. We can also use the same channel to stop our infinite poll loop, which would let us avoid `os.Exit()` as mentioned below.